tests/phpunit/AutoReview/IntegrationGroup/IntegrationGroupProvider.php

```diff
@@ -0,0 +1,191 @@
+     * it checks the source classes, but it is not excluded that a fixture file used in a test case
+     * contains I/O operations. In this scenario, the current implementation would miss out that one.
+     */
+    public static function provideIoTestCaseTuple(): Generator
+    {
+        if (null !== self::$ioTestCaseClassesTuple) {
+            yield from self::$ioTestCaseClassesTuple;
+
+            return;
+        }
+
+        self::$ioTestCaseClassesTuple = array_values(array_filter(array_map(
+            [self::class, 'checkIoOperations'],
+            iterator_to_array(ProjectCodeProvider::provideSourceClasses(), true)
+        )));
+
+        yield from self::$ioTestCaseClassesTuple;
+    }
+
+    public static function ioTestCaseTupleProvider(): Generator
+    {
+        yield from self::provideIoTestCaseTuple();
+    }
+
+    /**
+     * @return string[]|null
+     */
+    private static function checkIoOperations(string $className): ?array
+    {
+        $classReflection = new ReflectionClass($className);
+
+        $testCaseClass = SourceTestClassNameScheme::getTestClassName($className);
+
+        try {
+            $testCaseReflection = new ReflectionClass($testCaseClass);
+        } catch (ReflectionException $exception) {
+            // No test case could be find for this source file
+            return null;
+        }
+
+        //
+        // Check the test cases code
+        //
+        $testCaseFileName = $testCaseReflection->getFileName();
+        $testCaseCode = file_get_contents($testCaseFileName);
+
+        // Case where the test case itself use I/O functions
+        if (self::codeContainsIoFunctions($testCaseCode)) {
+            return [$testCaseClass, $testCaseFileName];
+        }
+
+        $parentTestCaseClassReflection = $testCaseReflection->getParentClass();
+
+        Assert::isInstanceOf($parentTestCaseClassReflection, ReflectionClass::class);
+
+        while ($parentTestCaseClassReflection->getName() !== TestCase::class) {
+            if ($parentTestCaseClassReflection->getName() === FileSystemTestCase::class) {
+                return [$testCaseClass, $parentTestCaseClassReflection->getFileName()];
+            }
+
+            $parentTestCaseFileName = $parentTestCaseClassReflection->getFileName();
+
+            $testCaseCode = file_get_contents($parentTestCaseFileName);
+
+            if (self::codeContainsIoFunctions($testCaseCode)) {
+                return [$testCaseClass, $parentTestCaseFileName];
+            }
+
+            $parentTestCaseClassReflection = $parentTestCaseClassReflection->getParentClass();
+
+            Assert::isInstanceOf($parentTestCaseClassReflection, ReflectionClass::class);
+        }
+
+        //
+        // Check the source class code
+        //
+        $classFileName = $classReflection->getFileName();
+        $classCode = file_get_contents($classFileName);
+
+        // Case where the test case itself use I/O functions
```

Comment:
seems like a copy-pasted wrong comment, as here we are checking source class' code, not the code of the test